### PR TITLE
Reacting to orientation changes

### PR DESCRIPTION
**Pull request by martinmitrevski**
### 🔗 Issue Link
[JIRA](https://stream-io.atlassian.net/browse/SWUI-107?atlOrigin=eyJpIjoiZWEwOTI5OTE3MmNlNDFiNDhlMjU0MmM0Yzc3YmVkYmEiLCJwIjoiaiJ9)

### 🎯 Goal

React to orientation changes, especially needed for iPad apps.

### 🛠 Implementation

SwiftUI doesn't have much support, so we need to manually react to orientation changes and redraw the view.

### 🧪 Testing

Open the app on iPad and test orientation changes.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

**Comment by martinmitrevski** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">There seems to be app changes but CHANGELOG wasn't modified.<br />
Please include an entry if the PR includes user-facing changes.<br />
You can find it at <a href="https://github.com/GetStream/stream-chat-swift/blob/main/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageContainerView.swift [hunk 1]
```diff
@@ -193,7 +193,12 @@ struct MessageContainerView<Factory: ViewFactory>: View {
     }
     
     private var spacerWidth: CGFloat {
-        (width ?? 0) / 4
+        let width = width ?? 0
+        if isIPad {
+            return 2 * width / 3
+        } else {
+            return width / 4
+        }
```
Comment (DaemonLoki) [inline Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageContainerView.swift:201]: A little confusing as to which width is which (tongue twister alarm 🚨)...
Maybe call the local constant of `width` something like `proposedViewWidth` or something that makes more sense?

File: Sources/StreamChatSwiftUI/ChatChannel/ChatChannelView.swift [hunk 2]
```diff
@@ -14,8 +14,14 @@ public struct ChatChannelView<Factory: ViewFactory>: View, KeyboardReadable {
     @State private var messageDisplayInfo: MessageDisplayInfo?
     @State private var keyboardShown = false
     @State private var tabBarAvailable: Bool = false
+    @State private var orientation = UIDevice.current.orientation
```
[the remark this answers is not in the document]
Comment (martinmitrevski) [reply Sources/StreamChatSwiftUI/ChatChannel/ChatChannelView.swift:17]: Actually the opposite. The id makes the view re-draw itself when orientation changes. The issue was that the `GeometryReader` falsely reported the old width when orientation changed, thus leading to the bug. Now, we always re-draw when the orientation is changed and the id enables that. There were no "more elegant" options to trigger that.

File: Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageContainerView.swift (hunk 1 above)
Comment (martinmitrevski) [reply Sources/StreamChatSwiftUI/ChatChannel/MessageList/MessageContainerView.swift:201]: Yes, makes sense, I'll update that.

**Review by DaemonLoki — APPROVED** [review]
LGTM

**Pull request merged**